addons/account_voucher/migrations/9.0.1.0/post-migration.py

```diff
@@ -71,9 +71,21 @@ def create_payments_from_vouchers(env):
         SET payment_id = av.id
         FROM account_voucher_line avl
         JOIN account_voucher av ON av.id = avl.voucher_id
-        WHERE avl.move_line_id=aml.id
+        WHERE avl.move_line_id = aml.id
         AND av.voucher_type IN ('receipt', 'payment')
-        AND av.state in ('draft', 'posted')
+        AND av.state IN ('draft', 'posted')
+        AND avl.reconcile
```

Comment:
This will only include the voucher lines whose corresponding move_id was reconciled with a payment.